core/src/main/java/com/orange/ouds/core/component/OudsBadge.kt

```diff
@@ -305,93 +358,196 @@ const val OudsBadgeMaxCount = 99
+
+        /**
+         * Creates an instance of [OudsBadgeWithIconStatus.Positive] with its default dedicated icon.
+         */
+        constructor(icon: OudsBadgeIcon.Default = OudsBadgeIcon.Default) : this(icon as OudsBadgeIcon)
+
+        @Composable
+        override fun getDefaultIconPainter() = painterResource(OudsTheme.drawableResources.success)
+    }
+
+    /**
+     * Provides informational context without urgency.
+     * Its [icon] is an [OudsBadgeIcon.Default].
+     */
+    class Info internal constructor(icon: OudsBadgeIcon) : OudsBadgeWithIconStatus(icon), BadgeFunctionalStatus {
+        /**
+         * Creates an instance of [OudsBadgeWithIconStatus.Info] with its default dedicated icon.
+         */
+        constructor(icon: OudsBadgeIcon.Default = OudsBadgeIcon.Default) : this(icon as OudsBadgeIcon)
+
+        @Composable
+        override fun getDefaultIconPainter() = painterResource(OudsTheme.drawableResources.information)
+    }
+
+    /**
+     * Negatives the user to potential risks or cautionary messages.
+     * Its [icon] is an [OudsBadgeIcon.Default].
+     */
+    class Warning internal constructor(icon: OudsBadgeIcon) : OudsBadgeWithIconStatus(icon), BadgeFunctionalStatus {
+        /**
+         * Creates an instance of [OudsBadgeWithIconStatus.Warning] with its default dedicated icon.
+         */
+        constructor(icon: OudsBadgeIcon.Default = OudsBadgeIcon.Default) : this(icon as OudsBadgeIcon)
+
+        @Composable
+        override fun getDefaultIconPainter() = painterResource(OudsTheme.drawableResources.warningExternalShape)
+    }
 
     /**
      * Draws attention to important or critical information.
      * Often used for errors, restrictions, or urgent messages, but not exclusively for failures.
      */
-    Negative,
-
     /**
-     * Indicate when the user isn’t allowed to interact with an element.
-     * They remove all interactivity from a text or icon elements.
+     * Draws attention to important or critical information.
+     * Often used for errors, restrictions, or urgent messages, but not exclusively for failures.
+     * Its [icon] is an [OudsBadgeIcon.Default].
      */
-    Disabled;
+    class Negative internal constructor(icon: OudsBadgeIcon) : OudsBadgeWithIconStatus(icon), BadgeFunctionalStatus {
+
+        /**
+         * Creates an instance of [OudsBadgeWithIconStatus.Negative] with its default dedicated icon.
+         */
+        constructor(icon: OudsBadgeIcon.Default = OudsBadgeIcon.Default) : this(icon as OudsBadgeIcon)
+
+        @Composable
+        override fun getDefaultIconPainter() = painterResource(OudsTheme.drawableResources.important)
+    }
 
     /**
      * The color associated with this status.
      */
-    val color: Color
-        @Composable
+    @Composable
+    fun color() = this.correspondingBadgeStatus.color()
+
+    internal val correspondingBadgeStatus
```

Comment:
What do you think about changing this property to a `toBadgeStatus()` method? This would be consistent with `toControlState()` that we already have in `OudsControlItemState`.